htmresearch/frameworks/pytorch/mnist_sparse_experiment.py

```diff
@@ -221,8 +221,11 @@ def iterate(self, params, repetition, iteration):
 
   def finalize(self, params, rep):
     """
-    Save the full model once we are done.
+    Called once we are done.
     """
+    self.pruneWeights(params)
```

Comment:
Could you make pruneWeights a method of the model itself? I might want to save the unpruned model, and then separately prune it for analysis.